compiler/src/dotty/tools/dotc/transform/TailRec.scala

```diff
@@ -174,12 +166,51 @@ class TailRec extends MiniPhase {
 
   }
 
-  class TailRecElimination(method: Symbol, enclosingClass: Symbol, isMandatory: Boolean, label: Symbol) extends tpd.TreeMap {
+  class TailRecElimination(method: Symbol, enclosingClass: Symbol, paramSyms: List[Symbol], isMandatory: Boolean) extends tpd.TreeMap {
 
     import dotty.tools.dotc.ast.tpd._
 
     var rewrote: Boolean = false
 
+    var continueLabel: Option[Symbol] = None
```

Comment:
It is weird to use an `Option` here since the value is not really optional. Here is the convention we tend to follow for idempotent lazily computed values:
```scala
private[this] myContinueLabel: Symbol = _
def continueLabel(implicit ctx: Context) = {
  if (myContinueLabel == null) {
    myContinueLabel = ...
  }
  myContinueLabel
}
```